src/WebJobs.Extensions.DurableTask.Analyzers/SyntaxNodeUtils.cs

```diff
@@ -308,133 +345,208 @@ private static bool IsSpecifiedAttribute(AttributeSyntax attribute, string attri
             return false;
         }
 
-        internal static string GetQualifiedTypeName(ITypeSymbol typeInfo)
+        public static bool IsDurableActivityContext(ITypeSymbol type)
         {
-            if (typeInfo != null)
+            if (type == null)
             {
-                if (typeInfo is INamedTypeSymbol namedTypeInfo)
-                {
-                    var tupleUnderlyingType = namedTypeInfo.TupleUnderlyingType;
-                    if (tupleUnderlyingType != null)
-                    {
-                        return $"({string.Join(", ", tupleUnderlyingType.TypeArguments.Select(x => x.ToString()))})";
-                    }
+                return false;
+            }
 
-                    return typeInfo.ToString();
-                }
+            return (type.ToString().Equals("Microsoft.Azure.WebJobs.Extensions.DurableTask.IDurableActivityContext")
+                || type.ToString().Equals("Microsoft.Azure.WebJobs.DurableActivityContext")
+                || type.ToString().Equals("Microsoft.Azure.WebJobs.DurableActivityContextBase"));
+        }
 
-                var arrayString = "";
-                if (typeInfo.Kind.Equals(SymbolKind.ArrayType))
-                {
-                    arrayString = "[]";
-                    typeInfo = ((IArrayTypeSymbol)typeInfo).ElementType;
-                }
+        public static bool IsMatchingDerivedOrCompatibleType(ITypeSymbol subclassOrMatching, ITypeSymbol superOrMatching)
```

Comment:
As you will see in the code below, I have subclassOrMatching and superOrMatching listed as parameters throughout most of these methods. The order of these parameters matters. An input to a function can be a subclass, but not the other way around. The expected return type can be a super, but not the other way around.

I dislike that this method would be so easy to misuse. Of course, this method is unlikely to get misused in this project but I wanted input on this approach.

What is a better way to handle cases where parameters have order that matter? I thought about making this an extension method, but ultimately kept it matching the other methods that follow for consistency. On second thought, I could always make the public method an extension method and the private methods I could keep with parameters that require the specific order. Thoughts?

Reply:
In general, I do think that an extension method would probably be the best approach for this comparison of two things of the same type, as then it is clear which is the "object" and which is the "subject".